clients/fides-js/src/lib/preferences.ts

```diff
@@ -117,13 +123,26 @@ export const updateConsentPreferences = async ({
   if (!updateCookie && !consentPreferencesToSave) {
     throw new Error("updateCookie is required");
   }
+  const trigger = {
+    ...(eventExtraDetails?.trigger as FidesEventDetailsTrigger),
+    origin:
+      (eventExtraDetails?.trigger as FidesEventDetailsTrigger)?.origin ||
+      (consentMethod === ConsentMethod.SCRIPT ||
+      consentMethod === ConsentMethod.GPC ||
+      consentMethod === ConsentMethod.OT_MIGRATION
+        ? "external"
+        : "fides"),
+  };
```

Comment:
automatically adds origin based on consent method, since all of these methods are triggered externally.

Reply:
1. What exactly are we calling "external"? Like not directly from a user action?
2. Can we make these groupings more programmatic? E.g. when adding a new consent method, I should easily be able to see that it needs to be put in a grouping of either "external" or "fides"

Reply:
2. Good idea, I'll make that update